packages/application-shell/src/components/application-shell/application-shell.js

```diff
@@ -282,42 +282,20 @@ export const RestrictedApplication = props => (
-
-                                  if (
-                                    props.environment.servedByProxy !== true &&
-                                    !previousProjectKey
-                                  )
-                                    return (
-                                      <Spacings.Stack scale="m">
-                                        <Text.Headline elementType="h2">
-                                          Development mode without projects
-                                        </Text.Headline>
-                                        <Text.Body>
-                                          You are using the Merchant Center in
-                                          development mode - it is not served by
-                                          a proxy. Moreover, you do not have any
-                                          projects yet. As a result we did not
-                                          redirect you anywhere (e.g. another
-                                          application).
-                                        </Text.Body>
-                                        <Text.Body>
-                                          Please go to the
-                                          `application-accounts` to create a
-                                          project first.
-                                        </Text.Body>
-                                        <LinkButton
-                                          to={`account/projects/new`}
-                                          label="Create project"
-                                        />
-                                      </Spacings.Stack>
-                                    );
+                                  /**
+                                   * NOTE:
+                                   *   Given the user has not been loaded a loading spinner is shown.
+                                   *   Given the user was not working on a project previously nor has a default
+                                   *   project she/she will be promted to create one.
+                                   *   Given the user was working in a project previously or has a default
+                                   *   project a redirect to it happens.
+                                   */
 
-                                  // NOTE: Given no previous `projectKey` is found we redirect to the base url.
+                                  if (!user) return <ApplicationLoader />;
+                                  if (!previousProjectKey)
+                                    return <RedirectToProjectCreate />;
                                   return (
-                                    <Redirect
-                                      to={`/${previousProjectKey || ''}`}
-                                    />
+                                    <Redirect to={`/${previousProjectKey}`} />
```

Comment:
@emmenko the bug was given you're behing the proxy and don't have a `projectKey` in the url (which is when we end up here) then we redirected to `/` instead of `account/projects/new`. Now we redirect there within the new component too (same as with the url containing a `projectKey`).